docs/extend/server-schemas.md

```diff
@@ -0,0 +1,64 @@
+# Server component schemas
+
+Starting with Zowe v2.0, each Component in Zowe must contain a [json schema](https://json-schema.org/) describing the configuration parameters that are valid for its component section in Zowe's server configuration. If a component does not have anything that can be configured, this file can just be boilerplate specifying that it fully inherits generic Component parameters and nothing more.
```

Comment:
the reason to require it is because making a schema with the boilerplate only takes a minute, but not having any enforcement will end up with people who would have been perfectly fine with making a nice schema not even knowing  it was possible to do.